tests/testthat/test-apply_theme.R

```diff
@@ -0,0 +1,407 @@
+  
+})
+
+testthat::test_that("T1.2 Not specifying any parameters returns a list.",{
+  
+  theme <- visR::define_theme()
+  
+  testthat::expect_true(is.list(theme))
+  
+})
+
+testthat::test_that("T1.3 No error when `strata` is `NULL`.",{
+  
+  testthat::expect_error(visR::define_theme(strata = NULL), NA)
+  
+})
+
+testthat::test_that("T1.4 A warning when `strata` is an empty `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(strata = list()))
+  
+})
+
+testthat::test_that("T1.5 A warning when `strata` is an unnamed `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(strata = list("v", "i", "s", "R")))
+  
+})
+
+testthat::test_that("T1.6 No warning when `strata` is a named `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(strata = list("visR" = "visR")), NA)
+  
+})
+
+testthat::test_that("T1.7 No error when `fontsizes` is `NULL`.",{
+  
+  testthat::expect_error(visR::define_theme(fontsizes = NULL), NA)
+  
+})
+
+testthat::test_that("T1.8 A warning when `fontsizes` is an empty `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(fontsizes = list()))
+  
+})
+
+testthat::test_that("T1.9 A warning when `fontsizes` is an unnamed `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(fontsizes = list("v", "i", "s", "R")))
+  
+})
+
+testthat::test_that("T1.10 No warning when `fontsizes` is a named `list`.",{
+  
+  testthat::expect_warning(visR::define_theme(fontsizes = list("visR" = "visR")), NA)
+  
+})
+
+testthat::test_that("T1.11 A message when `fontsizes` is a numerical value.",{
+  
+  testthat::expect_message(visR::define_theme(fontsizes = 12))
+  
+})
+
+testthat::test_that("T1.12 A warning when `fontsizes` is neither `NULL`, a `list` or a `numeric`.",{
+  
+  testthat::expect_warning(visR::define_theme(fontsizes = "visR"))
+  
+})
+
+testthat::test_that("T1.13 No error when `fontfamily` is a string.",{
+  
+  testthat::expect_error(visR::define_theme(fontfamily = "Times"), NA)
+  
+})
+
+testthat::test_that("T1.14 A warning when `fontfamily` is anything but a string.",{
+  
+  testthat::expect_warning(visR::define_theme(fontfamily = NULL))
+  testthat::expect_warning(visR::define_theme(fontfamily = 12))
+  testthat::expect_warning(visR::define_theme(fontfamily = TRUE))
+  testthat::expect_warning(visR::define_theme(fontfamily = c()))
+  
+})
+
+testthat::test_that("T1.15 No error when `grid` is a boolean",{
+  
+  testthat::expect_error(visR::define_theme(grid = TRUE), NA)
+  testthat::expect_error(visR::define_theme(grid = FALSE), NA)
+  
+})
+
+testthat::test_that("T1.16 A warning when `grid` is anything but a boolean",{
+  
+  testthat::expect_warning(visR::define_theme(grid = NULL))
+  testthat::expect_warning(visR::define_theme(grid = 12))
+  testthat::expect_warning(visR::define_theme(grid = "visR"))
+  testthat::expect_warning(visR::define_theme(grid = c()))
+  
+})
+
+testthat::test_that("T1.17 No error when `bg` is a character.",{
```

Comment:
any kind of character? or a valid color string?

Reply:
Right now, any kind of character. We pass this through to ggplot and let them figure out what a valid colour is.